datasets/synthetic_dataset.py

```diff
@@ -0,0 +1,160 @@
+from dataclasses import dataclass, field
+
+import torch
+from transformers import HfArgumentParser
+
+from diarizers import SyntheticDataset, SyntheticDatasetConfig
+
+
+@dataclass
+class ASRDatasetArguments:
```

Comment:
Wondering whether we need both `ASRDatasetArguments` **and** `SyntheticDatasetConfig`? IMO there's some unnecessary redundancy here -> we only need one of the two config classes.

I would probably stick with `SyntheticDatasetConfig` to make it easy, and have users define the args in a python cell:
```python
from diarizers import SyntheticDatasetConfig, SyntheticDataset

synthetic_config = SyntheticDatasetConfig(
        dataset_name=...,
        subset=...,
        split=...,
...
)

dataset = SyntheticDataset(synthetic_config).generate(batch_size=..., num_proc=...)
```

Reply:
If I understand well you propose to remove the `synthetic_pipeline.py` script from the repository and propose instead a python code snippet using `SyntheticDatasetConfig` in the readme ?